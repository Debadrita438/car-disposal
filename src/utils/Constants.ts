export default {
  baseUrlApi: 'https://hickman-hybrid.technoexponent.net/api/',

  // api endpoints
  register: 'register',
  verifyOtp: 'verify-register',
  login: 'login',
  forgetPassword: 'forget-password',
  resetPassword: 'reset-password',
  resendOtp: 'resend-otp',
  logout: 'logout',
  userDetails: 'user/',
  forgotPassEmailVerify: 'verfigy-forget-password',
  updateProfile: 'user/update/',
  updateProfileImage: 'user/update-profile-picture/',
  changePassword: 'user/change-password/',
};
