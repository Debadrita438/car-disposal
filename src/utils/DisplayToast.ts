import Toast, {ToastType} from 'react-native-toast-message';
import {Colors, Fonts} from '../theme';

export const displayToast = (message: string, messageType: ToastType) => {
  Toast.show({
    type: messageType,
    text1: message,
    text1Style: {
      fontFamily: Fonts.regular,
      color: Colors.blackColor,
      fontSize: 12,
      fontWeight: '400',
    },
  });
};
