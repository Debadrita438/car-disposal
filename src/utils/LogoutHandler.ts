import {resetChangePassword} from '../redux/App/ChangePasswordSlice';
import {resetUpdateProfile} from '../redux/App/EditAccount/EditProfileSlice';
import {resetUpdateProfileImage} from '../redux/App/EditAccount/UpdateProfileImageSlice';
import {resetLogout} from '../redux/App/LogoutSlice';
import {resetUserDetails} from '../redux/App/UserDetailsSlice';
import {AppDispatch} from '../redux/store';
import {storage} from './Storage';

export const handleLogout = (dispatch: AppDispatch, navigation: any) => {
  // Dispatch reset actions
  dispatch(resetLogout());
  dispatch(resetUserDetails());
  dispatch(resetChangePassword());
  dispatch(resetUpdateProfile());
  dispatch(resetUpdateProfileImage());

  // Delete storage items
  storage.delete('token');
  storage.delete('userId');
  storage.delete('remember');
  storage.delete('email');
  storage.delete('password');

  // Navigate to login screen
  navigation.replace('LoginScreen');
};
