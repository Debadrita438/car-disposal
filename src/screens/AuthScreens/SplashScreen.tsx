import React, {useEffect} from 'react';
import {View, StyleSheet, Image} from 'react-native';
import {storage} from '../../utils/Storage';

const SplashScreen = ({navigation}) => {
  useEffect(() => {
    // Simulate an async action like fetching data
    const timeout = setTimeout(() => {
      // Navigate to the main screen after 3 seconds
      if (storage.contains('token')) {
        navigation.replace('AppScreen');
      } else {
        navigation.replace('IntroScreen'); // or your main app screen
      }
    }, 3000);

    return () => clearTimeout(timeout); // Clean up the timer
  }, [navigation]);

  return (
    <View style={styles.container}>
      <Image
        source={require('../../assets/images/splashScreen.png')}
        style={styles.logo}
        resizeMode="stretch"
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#fff', // Add your background color here
  },
  logo: {
    width: '100%',
    height: '100%', // Adjust the size of your logo
  },
  text: {
    fontSize: 20,
    marginTop: 20,
    fontWeight: 'bold',
  },
});

export default SplashScreen;
