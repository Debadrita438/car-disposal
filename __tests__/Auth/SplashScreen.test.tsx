import React from 'react';
import {render, act} from '@testing-library/react-native';
import SplashScreen from '../../src/screens/AuthScreens/SplashScreen';
import renderer from 'react-test-renderer';
import {storage} from '../../src/utils/Storage';

// Mock the navigation prop
jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({
    replace: jest.fn(),
  }),
}));

// Mock the storage module (MMKV)
jest.mock('../../src/utils/Storage', () => ({
  storage: {
    contains: jest.fn(),
  },
}));

describe('SplashScreen', () => {
  const mockNavigation = {replace: jest.fn()};

  beforeEach(() => {
    jest.clearAllMocks(); // Reset mocks before each test
  });

  it('navigates to AppScreen if the token exists', () => {
    // Mock storage.contains to return true (simulate token exists)
    (storage.contains as jest.Mock).mockReturnValue(true);

    jest.useFakeTimers(); // Mock timers
    render(<SplashScreen navigation={mockNavigation} />);

    // Fast-forward until all timers have been executed
    act(() => {
      jest.runAllTimers();
    });

    expect(mockNavigation.replace).toHaveBeenCalledWith('AppScreen');
  });

  it('navigates to IntroScreen if the token does not exist', () => {
    // Mock storage.contains to return false (simulate no token)
    (storage.contains as jest.Mock).mockReturnValue(false);

    jest.useFakeTimers();
    render(<SplashScreen navigation={mockNavigation} />);

    act(() => {
      jest.runAllTimers();
    });

    expect(mockNavigation.replace).toHaveBeenCalledWith('IntroScreen');
  });

  it('renders the splash screen image', () => {
    const {getByTestId} = render(<SplashScreen navigation={mockNavigation} />);
    const splashImage = getByTestId('splash-image'); // Ensure testID is added to the Image component

    expect(splashImage).toBeTruthy();
  });
});

test('renders correctly', () => {
  const mockNavigation = {replace: jest.fn()};
  const tree = renderer
    .create(<SplashScreen navigation={mockNavigation} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
