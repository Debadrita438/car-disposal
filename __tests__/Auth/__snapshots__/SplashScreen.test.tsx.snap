// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    resizeMode="stretch"
    source={
      {
        "testUri": "../../../src/assets/images/splashScreen.png",
      }
    }
    style={
      {
        "height": "100%",
        "width": "100%",
      }
    }
    testID="splash-image"
  />
</View>
`;
